internal/database/connectors/dbconn.go

```diff
@@ -66,7 +66,12 @@ func NewPostgresDB(opts ...DBOption) (*PostgresDB, error) {
 }
 
 func (p *PostgresDB) Connect(ctx context.Context) (err error) {
-	dsn := fmt.Sprintf("host=%s user=%s dbname=%s port=%s sslmode=require", p.Host, p.User, p.Name, p.Port)
+	var dsn string
+	if len(p.Password) > 0 {
```

Comment:
@abhishekp03 , it is safer to add not nil check here `if p.Password != nil && len(*p.Password) > 0`